src/Shared/FileMatcher.cs

```diff
@@ -2689,5 +2690,26 @@ private static bool DirectoryEndsWithPattern(string directoryPath, string patter
         };
 
         internal static bool IsRecursiveDirectoryMatch(string path) => path.TrimTrailingSlashes() == recursiveDirectoryMatch;
+
+        public struct GetFilesImplementationResults
```

Comment:
This is the results struct mentioned in the description, I have a few questions regarding this:
1. Should I pass results in a different way?
2. Should I reduce the number of items in this struct (they seem necessary to me, but I'm not certain if anything could be optimized further here)?
3. I use Default wherever necessary in various codepaths since repeating constructors seemed redundant in certain areas. Should it be removed?